2DClothSim.cpp

```diff
@@ -3,30 +3,99 @@
 
 #include "2DClothSim.h"
 
-using namespace std;
-
 int main(int argc, char* argv[])
 {
     SDL_Init(SDL_INIT_VIDEO);
 
-    SDL_Window* window = SDL_CreateWindow(
-        "SDL2Test",
+    ClothSim::window = SDL_CreateWindow(
+        "2DClothSim",
         SDL_WINDOWPOS_UNDEFINED,
         SDL_WINDOWPOS_UNDEFINED,
         640,
         480,
         0
     );
 
-    SDL_Renderer* renderer = SDL_CreateRenderer(window, -1, SDL_RENDERER_SOFTWARE);
-    SDL_SetRenderDrawColor(renderer, 0, 0, 0, SDL_ALPHA_OPAQUE);
-    SDL_RenderClear(renderer);
-    SDL_RenderPresent(renderer);
-
-    SDL_Delay(3000);
+    ClothSim::renderer = SDL_CreateRenderer(ClothSim::window, -1, SDL_RENDERER_SOFTWARE);
+    ClothSim::Cloth* cloth = new ClothSim::Cloth();
+    SDL_Event e;
+    do {
+        SDL_SetRenderDrawColor(ClothSim::renderer, 0xFF, 0xFF, 0xFF, SDL_ALPHA_OPAQUE);
+        SDL_RenderClear(ClothSim::renderer);
+        ClothSim::update(cloth, 0.002f);
+        SDL_SetRenderDrawColor(ClothSim::renderer, 0x00, 0x00, 0xFF, SDL_ALPHA_OPAQUE);
+        SDL_RenderDrawPointsF(ClothSim::renderer, cloth->getVerts()->data(), cloth->getPixelsX() * cloth->getPixelsY());
+        SDL_RenderPresent(ClothSim::renderer);
+        SDL_PollEvent(&e);
+    } while (e.type != SDL_QUIT);
 
-    SDL_DestroyWindow(window);
+    SDL_DestroyWindow(ClothSim::window);
     SDL_Quit();
 
     return 0;
 }
+
+void ClothSim::Cloth::initVerts() {
+    verts = new std::vector<SDL_FPoint>;
+    velos = new std::vector<struct vect2D>;
+    accs = new std::vector<struct vect2D>;
+
+    for (int64_t i = 0; i < pixels_y; i++) {
+        for (int64_t j = 0; j < pixels_x; j++) {
+            verts->push_back(SDL_FPoint());
+            velos->push_back(vect2D(.0f, .0f));
+            accs->push_back(vect2D(.0f, 9.8f));
+            (*verts)[(i * pixels_x) + j].x = (float)(10 + 2 * j);
```

Comment:
Why not just push_back this????

Reply:
`SDL_FPoint` does not have such a constructor. I need to create an empty point and then populate its coordinates